kt_jvm_grpc.bzl

```diff
@@ -0,0 +1,145 @@
+
+    gen_src_dir_name = "%s/ktgrpc" % name
+    gen_src_dir = ctx.actions.declare_directory(gen_src_dir_name)
+    source_jar = ctx.actions.declare_file("%s.srcjar" % name)
+
+    _invoke_generator(ctx, proto_dep, gen_src_dir)
+    _build_srcjar(ctx, proto_dep, gen_src_dir, source_jar)
+
+    java_info = ctx.attr.deps[0][JavaInfo]
+    default_info = DefaultInfo(files = depset([source_jar, gen_src_dir]))
+
+    return [java_info, default_info]
+
+_kt_grpc_generate_code = rule(
+    attrs = dict(
+        srcs = attr.label_list(
+            providers = [ProtoInfo],
+        ),
+        deps = attr.label_list(
+            providers = [JavaInfo],
+        ),
+        _zipper = attr.label(
+            executable = True,
+            cfg = "host",
+            default = Label("@bazel_tools//tools/zip:zipper"),
+            allow_files = True,
+        ),
+        _generator = attr.label(
+            default = Label("//compiler/src/main/java/io/grpc/kotlin/generator:GeneratorRunner"),
+            cfg = "host",
+            executable = True,
+        ),
+    ),
+    fragments = ["java"],
+    provides = [JavaInfo],
+    implementation = _kt_grpc_generate_code_impl,
+)
+
+def kt_jvm_grpc_library(
+        name,
+        srcs = None,
+        deps = None,
+        tags = None,
+        testonly = None,  # None to not override Blaze's default for //javatests, b/112708042
+        compatible_with = None,
+        restricted_to = None,
+        visibility = None,
+        flavor = None,
+        deprecation = None,
+        features = []):
+    """This rule compiles Kotlin APIs for gRPC services from the proto_library in srcs.
+
+    Args:
+      name: A name for the target
+      srcs: Exactly one proto_library target to generate Kotlin APIs for
+      deps: Exactly one java_grpc_library target for srcs[0]
+      tags: A list of string tags passed to generated targets.
+      testonly: Whether this target is intended only for tests.
+      compatible_with: Standard attribute, see http://go/be-common#common.compatible_with
+      restricted_to: Standard attribute, see http://go/be-common#common.restricted_to
+      visibility: A list of targets allowed to depend on this rule.
+      flavor: "normal" (default) for normal proto runtime, or "lite" for the lite runtime
+        (for Android usage)
+      deprecation: Standard attribute, see http://go/be-common#common.deprecation
+      features: Features enabled.
+    """
+
+    srcs = srcs or []
+    deps = deps or []
+
+    if len(srcs) != 1:
+        fail("Expected exactly one src", "srcs")
+    if len(deps) != 1:
+        fail("Expected exactly one dep", "deps")
+
+    if flavor == None or flavor == "normal":
+        deps.extend([
+            "@com_github_grpc_grpc_kotlin//stub/src/main/java/io/grpc/kotlin:stub",
+            "@com_github_grpc_grpc_kotlin//stub/src/main/java/io/grpc/kotlin:context",
+        ])
+    elif flavor == "lite":
+        fail("Android support is unimplemented")
```

Comment:
How difficult is it to add lite support, now that this has been implemented in `gradle`? And is this a priority?

Reply:
I took out the flavor check here since I don't think it matters for the Kotlin stubs and we pass the flavor to java_grpc_library which I think should make this work for Android.